apis/projects/v1alpha1/referencers.go

```diff
@@ -119,6 +119,29 @@ func (mg *DeployToken) ResolveReferences(ctx context.Context, c client.Reader) e
 	return nil
 }
 
+// ResolveReferences of this Deploy Key
+func (mg *DeployKey) ResolveReferences(ctx context.Context, c client.Reader) error {
```

Comment:
To save you time and complexity I recommend to use `angryjet` to generate the `ResolveReferences` for your types, like here: https://github.com/crossplane-contrib/provider-aws/blob/master/apis/eks/manualv1alpha1/nodegroup_types.go#L65

Generation is already included in `make generate`.